.github/workflows/setup-homebrew.yml

```diff
@@ -11,15 +11,42 @@ on:
 jobs:
   setup:
     strategy:
+      fail-fast: false
       matrix:
-        os:
-          - ubuntu-latest
-          - ubuntu-22.04
-          - macOS-latest
-    runs-on: ${{matrix.os}}
+        include:
+          - os: macos-latest
+            key: macos
+          - os: ubuntu-18.04
+            key: ubuntu-18
+          - os: ubuntu-22.04
+            key: ubuntu-22
+          - os: ubuntu-22.04
+            key: docker-root
+            container:
+              image: ghcr.io/homebrew/ubuntu22.04:master
+              options: --user=root
+          - os: ubuntu-22.04
+            key: docker-linuxbrew
+            container:
+              image: ghcr.io/homebrew/ubuntu22.04:master
+              options: --user=linuxbrew
+    runs-on: ${{ matrix.os }}
+    container: ${{ matrix.container }}
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        run: |
+          # Allow writing under linuxbrew user
+          orig_uid=$(ls -nd $GITHUB_WORKSPACE | awk 'NR==1 {print $3}')
+          sudo chown -R $(whoami) $GITHUB_WORKSPACE $HOME
+
+          git config --global --add safe.directory $GITHUB_WORKSPACE
+          git init
+          git remote add origin $GITHUB_SERVER_URL/$GITHUB_REPOSITORY
+          git fetch --no-tags --depth=1 origin $GITHUB_SHA
+          git checkout --force $GITHUB_SHA
+
+          # Reset ownership so we test if the equivalent in setup-homebrew works.
+          sudo chown -R $orig_uid $GITHUB_WORKSPACE $HOME
```

Comment:
Can you explain a bit more what's going on here? I'm pretty lost.

Reply:
We never tested setup-homebrew under Homebrew Docker images in this repo's CI, so I thought it was about time we did.

`actions/checkout@v3` does not work with our Homebrew Docker images. The necessary permission fixes are included in our setup-homebrew, but the point of this CI is to checkout so we can test the setup-homebrew changes.

So this replaces `actions/checkout` with a simple checkout + the temporary permission change necessary for that to happen.

Reply:
> `actions/checkout@v3` does not work with our Homebrew Docker images. The necessary permission fixes are included in our setup-homebrew, but the point of this CI is to checkout so we can test the setup-homebrew changes.
> 
> So this replaces `actions/checkout` with a simple checkout + the temporary permission change necessary for that to happen.

Could this get added as an inline comment? Thanks ❤️